### PR TITLE
[Diagnostics] Reword malformed swiftsourceinfo diagnostic

### PR DESCRIPTION
**Pull request by bnbarham**
The original wording is not helpful to a user if they don't know what a
swiftsourceinfo actually is. Change the wording to provide some more
context, ie. that it is an unstable format and should not be
distributed.

Resolves rdar://80785874

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please smoke test

**Comment by bnbarham** [comment]
@swift-ci please smoke test

**Comment by bnbarham** [comment]
@swift-ci please smoke test

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -812,7 +812,9 @@ ERROR(serialization_allowing_error_type,none,
       "allowing deserialization of error type '%0' in module '%1'",
       (StringRef, StringRef))
 WARNING(serialization_malformed_sourceinfo,none,
-      "unable to use malformed module source info '%0'", (StringRef))
+      "'%0' is either malformed or generated by a different Swift version. "
+      "Note that it is uses an unstable format and may leak internal project "
```
Comment (akyrtzi) [inline include/swift/AST/DiagnosticsSema.def:816]: typo, extra 'is'

Comment (bnbarham) [reply include/swift/AST/DiagnosticsSema.def]: Argh, thanks. I originally had put up "is" and decided to change it to "uses". Not sure how I missed that I hadn't replaced it!

**Comment by bnbarham** [comment]
@swiftci please smoke test

**Comment by bnbarham** [comment]
@swift-ci please smoke test

**Review by nkcsgexi — APPROVED** [review]
Thank you!

**Review by akyrtzi — APPROVED** [review]

**Pull request merged**